gateway/ln-gateway/src/lib.rs

```diff
@@ -1351,7 +1328,13 @@ impl Gateway {
             }
 
             for (federation_id, federation_config) in federations {
-                if let Some(client) = self.clients.read().await.get(federation_id) {
+                if let Some(client) = self
+                    .federation_manager
+                    .read()
+                    .await
+                    .borrow_clients()
```

Comment:
IMO there should be a `get_client_by_id` function rather than borrowing the entire clients map

Reply:
There's a few places where this would be useful

Reply:
Done.